lib/cartopy/img_transform.py

```diff
@@ -285,7 +285,15 @@ def regrid(array, source_x_coords, source_y_coords, source_cs, target_proj,
                                            target_x_points.flatten(),
                                            target_y_points.flatten())
 
-    kdtree = scipy.spatial.cKDTree(xyz)
+    # Versions of scipy >= v0.16 added the balanced_tree argument,
+    # hence the handling of the specific exception. 
+    try:
+        kdtree = scipy.spatial.cKDTree(xyz, balanced_tree=False)
+    except TypeError as err:
+        if err.message[-14:-1] == 'balanced_tree':
```

Comment:
I don't think we need to check the contents of the error message. We can just go straight into doing the call, and if we get another TypeError exception, so be it.